builder/builder-source-archive.c

```diff
@@ -234,6 +237,19 @@ get_download_location (BuilderSourceArchive *self,
       return FALSE;
     }
 
+  sources_dirs = builder_context_get_sources_dirs (context);
+  for (i = 0; sources_dirs != NULL && i < sources_dirs->len; i++)
+    {
+      GFile* sources_root = g_ptr_array_index (sources_dirs, i);
+      g_autoptr(GFile) local_download_dir = g_file_get_child (sources_root, "downloads");
+      g_autoptr(GFile) local_sha256_dir = g_file_get_child (local_download_dir, self->sha256);
+      g_autoptr(GFile) local_file = g_file_get_child (local_sha256_dir, base_name);
+      if (g_file_query_exists (local_file, NULL)) {
+        *is_local = TRUE;
+        return g_steal_pointer (&local_file);
+      }
+    }
+
```

Comment:
is_local is set to FALSE right before get_download_location is called - so that case should be covered and we only need to set it to true when we have extra-sources, then we set it to TRUE.